code/datums/components/wall_mounted.dm

```diff
@@ -71,6 +71,9 @@
 	SIGNAL_HANDLER
 	var/obj/hanging_parent = parent
 
+	if(hanging_parent.obj_flags & NO_DECONSTRUCTION)
+		return
```

Comment:
This still needs to be reverted

...Ideally the `.deconstruct` call would handle `NO_DECONSTRUCTION` handling as it wants, but there's a shocking lack of consistency around `NO_DECONSTRUCTION`, where some objects treat it as "allow deconstruction, but make it drop no parts" and others simply "disallow deconstruction at all"

Reply:
This is what's fixing the cameras, if something has ``NO_DECONSTRUCTION`` flag then we shouldn't let it be deconstructed from it falling off of the wall. If something is left hanging over nothing then we should fix it case-by-case, if something is set to no_deconstruct then it should not deconstruct.

Reply:
Then you disagree with a lot of assumptions made by the code base, which implies this isn't a good fix (at least, not right now)

Reply:
What do you suggest would be a better fix here because I don't really agree.